bmb_vision/src/bmb_vision/rail_detection_helper.py

```diff
@@ -1,17 +1,289 @@
 import rospy
 from bmb_msgs.msg import RailDetection
 from sensor_msgs.msg import Image
-
+import numpy as np
+import cv2
 
 class RailDetectionHelper:
     def __init__(self):
         self.raw_image_sub = rospy.Subscriber("/rectified_image", Image, self.image_callback)
         self.rail_detection_pub = rospy.Publisher("/rail_detection", RailDetection, queue_size=1)
+    
+    def Distanceperp(x1, x3, height, m1, b1, m2, b2):
+        if m1 == 0 and m2 == 0:
+            distanceperp = abs(x3 - x1)
+        else:
+            if m1 != 0:
+                xone = 0 / m1 + b1
+                xtwo = 200 / m1 + b1
+                angle = np.arctan(height / (abs(xone - xtwo)))
+                x1 = 50 / m1 + b1
+            else:
+                x1 = x1
+            if m2 != 0:
+                xthree = 0 / m2 + b2
+                xfour = 200 / m2 + b2
+                angle = np.arctan(height / (abs(xthree - xfour)))
+                x2 = 50 / m2 + b2
+            else:
+                x2 = x3
+            distanceperp = abs(x2 - x1) * np.sin(angle)
+        return distanceperp
+
+
+    def equationoflineVector(x1, y1, x2, y2):
+        if (x2 - x1) == 0:
+            m = 0
+            b = 0
+        else:
+            m = (y2 - y1) / (x2 - x1)
+            b = x2 - y2 / m
+        return m, b
+        # x=t
+        # y=(m(t-b))
+
+
+    def ccw(A, B, C):
+        return (C[1] - A[1]) * (B[0] - A[0]) > (B[1] - A[1]) * (C[0] - A[0])
+
+
+    # Return true if line segments AB and CD intersect
+    def intersect(A, B, C, D):
+        return ccw(A, C, D) != ccw(B, C, D) and ccw(A, B, C) != ccw(A, B, D)
+
+    def gauge_measurement(m1, b1, m2, b2):
+        y = 0
+        gauge = 0
+        while y < 100:
+            x1 = y / m1 + b1
+            x2 = y / m2 + b2
+            gauge += x2 - x1
+            y += 1
+        meangauge = abs(gauge / 100)
+        print(f'Gauge:{meangauge}')
+
+
+    def railoffsetimagecrop(img, x0i, x0f, x1i, x1f):
```

Comment:
Use consistent naming! Name functions and variables like this: `my_cool_name`. Also run code formatting over the entire file (Ctrl+Alt+L).
Also, you code should be easy to read and self documenting for the most part. Right now I can't make sense of anything.